server/routes/applications.ts

```diff
@@ -0,0 +1,442 @@
+        return res.status(401).json({ message: 'requested application is not attached to the user.' });
+
+    if(!req.body)
+        return res.status(400).json({ message: 'no body provided for contact creation.' });
+
+    let company: string, position: string, location: string, jobPostUrl: string, description: string, salary: number, cardColor: string, progress: number;
+    let newDataObject: [id: string], any = {};
+    try {
+        if(company) {
+            if(!checkNonEmptyString(company))
+                throw new Error('Company must be a non-empty string.');
+            else
+                newDataObject['company'] = company;
+        }
+        if(position) {
+            if(!checkNonEmptyString(position))
+                throw new Error('Position must be a non-empty string.');
+            else
+                newDataObject['position'] = position;
+        }
+        if(location) {
+            if(!checkNonEmptyString(location))
+                throw new Error('Location must be a non-empty string.');
+            else
+                newDataObject['location'] = location;
+        }
+        if(jobPostUrl) {
+            if(!checkNonEmptyString(jobPostUrl))
+                throw new Error('Job Post URL must be a non-empty string.');
+            else
+                newDataObject['jobPostUrl'] = jobPostUrl;
+        }
+        if(description) {
+            if(!checkNonEmptyString(description))
+                throw new Error('Description must be a non-empty string.');
+            else
+                newDataObject['description'] = description;
+        }
+        if(salary) {
+            if(!checkPositiveNumber(salary))
+                throw new Error('Salary must be a positive number.');
+            else
+                newDataObject['salary'] = salary;
+        }
+        if(cardColor) {
+            if(!checkNonEmptyString(cardColor))
+                throw new Error('Card Color must be a non-empty string.');
+            else
+                newDataObject['cardColor'] = cardColor;
+        }
+        if(progress) {
+            if(!checkPositiveNumber(progress))
+                throw new Error('Progress must be a positive number.');
+            else
+                newDataObject['progress'] = progress;
+        }
+    } catch(e) {
+        return res.status(400).json({ message: e.message });
+    }
+
+    if(Object.keys(newDataObject).length === 0)
+        return res.status(400).json({ message: 'no fields provided to be changed.' });
+
+    try {
+        const application: ApplicationObject = 
+                            await updateApplication(
+                                req.session.user._id, applicationId,
+                                newDataObject['company'], newDataObject['position'],
+                                newDataObject['location'], newDataObject['jobPostUrl'],
+                                newDataObject['description'], newDataObject['salary'],
```

Comment:
I think we decided that the salary might not be known on creation so we won't pass it in

Reply:
This is an update function, not creation function